src/write.c

```diff
@@ -1208,8 +1208,10 @@ static avifResult avifValidateGrid(uint32_t gridCols,
     const avifImage * bottomRightCell = cellImages[cellCount - 1];
 #if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
     if (validateGainMap) {
-        firstCell = firstCell->gainMap.image;
-        bottomRightCell = bottomRightCell->gainMap.image;
+        assert(firstCell->gainMap && firstCell->gainMap->image);
```

Comment:
Just wondering if that can never happen and whether  AVIF_CHECKRES should be used

Reply:
No the caller should only call the function with validateGainMap if there is a gain map. I think defensive programming is not useful in this instance.